plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConnector.java

```diff
@@ -48,14 +49,17 @@ public PinotConnector(
             PinotSplitManager splitManager,
             PinotPageSourceProvider pageSourceProvider,
             PinotNodePartitioningProvider partitioningProvider,
-            PinotSessionProperties pinotSessionProperties)
+            PinotSessionProperties pinotSessionProperties,
+            PinotTableProperties tableProperties)
     {
         this.lifeCycleManager = requireNonNull(lifeCycleManager, "lifeCycleManager is null");
         this.metadata = requireNonNull(metadata, "metadata is null");
         this.splitManager = requireNonNull(splitManager, "splitManager is null");
         this.pageSourceProvider = requireNonNull(pageSourceProvider, "pageSourceProvider is null");
         this.partitioningProvider = requireNonNull(partitioningProvider, "partitioningProvider is null");
         this.sessionProperties = requireNonNull(pinotSessionProperties, "pinotSessionProperties is null");
+        requireNonNull(tableProperties, "tableProperties is null");
+        columnProperties = tableProperties.getColumnProperties();
```

Comment:
nit: Add `requireNonNull` and `ImmutableList.copyOf`.

Reply:
requireNonNull is on the line above, should I merge it?

Reply:
Let's add only `ImmutableList.copyOf`.